client/views/room/webdav/SaveToWebdavModal.tsx

```diff
@@ -0,0 +1,123 @@
+import { Modal, Box, ButtonGroup, Button, FieldGroup, Field, Select, SelectOptions, Throbber } from '@rocket.chat/fuselage';
+import { useUniqueId } from '@rocket.chat/fuselage-hooks';
+import React, { ReactElement, useState, useMemo } from 'react';
+import { useForm, Controller } from 'react-hook-form';
+
+import type { MessageAttachment } from '../../../../definition/IMessage/MessageAttachment/MessageAttachment';
+import type { IWebdavAccount } from '../../../../definition/IWebdavAccount';
+import { useMethod } from '../../../contexts/ServerContext';
+import { useToastMessageDispatch } from '../../../contexts/ToastMessagesContext';
+import { useTranslation } from '../../../contexts/TranslationContext';
+import { useEndpointData } from '../../../hooks/useEndpointData';
+import { getWebdavServerName } from '../../../lib/getWebdavServerName';
+
+type SaveToWebdavModalProps = {
+	onClose: () => void;
+	data: {
+		attachment: MessageAttachment;
+		url: string;
+	};
+};
+
+const SaveToWebdavModal = ({ onClose, data }: SaveToWebdavModalProps): ReactElement => {
+	const t = useTranslation();
+	const [isLoading, setIsLoading] = useState(false);
+	const dispatchToastMessage = useToastMessageDispatch();
+	const uploadFileToWebdav = useMethod('uploadFileToWebdav');
+	const accountIdField = useUniqueId();
+
+	const {
+		control,
+		handleSubmit,
+		formState: { errors },
+	} = useForm<{ accountId: string }>();
+
+	const { value } = useEndpointData('webdav.getMyAccounts');
+
+	const accountsOptions: SelectOptions = useMemo(() => {
+		if (value?.accounts) {
+			return value.accounts.map(({ _id, ...current }) => [_id, getWebdavServerName(current)]);
+		}
+
+		return [];
+	}, [value?.accounts]);
+
+	const handleSaveFile = ({ accountId }: { accountId: IWebdavAccount['_id'] }): void => {
+		setIsLoading(true);
+
+		const {
+			url,
+			attachment: { title },
+		} = data;
+
+		const fileRequest = new XMLHttpRequest();
```

Comment:
We need a way to close this request if the user wants to. I suggest a cancel button to close the modal and a useEffect to cancel the request on unMount. I think you can save the request in a ref so you can do this.

Reply:
We agreed to skip this for now!